WIP: Update configs with VBF multicategories

francescobrivio opened the pull request:
Changes done:
- Updating the configs to include VBF multicategories and prepare limit computation.
- Updated bTag categories (needed after HHbtag was introduced)
-  Added up/down variables in all the `main*_limit.cfg` and their binning in `selections*.cfg`

Still to do:
- update `DNNoutSM_kl_1` binning
- add missing systematics

francescobrivio commented:
@jaimeleonh could you have a look at the VBF multiclass binnings? The binnings are defined at the end of the `selectionCfg_*Tau_Legacy201*.cfg` files. 
I used the mpp values, it should be correct, maybe @riga can you confirm if it's ok?

jaimeleonh commented:
Hi @francescobrivio , if I understand correctly, same binning will we applied whether or not the multicategories are applied, right? Binnings are a bit different in both cases, but well, let's see. 

Apart from this, I've provided you only the binnings for the 2016 data and I see you have applied them to all years. Do you want me to run it for the other years as well?

francescobrivio commented:
> Hi @francescobrivio , if I understand correctly, same binning will we applied whether or not the multicategories are applied, right? Binnings are a bit different in both cases, but well, let's see.

Correct, here i'm only applying the mpp binning and the idea is to use each output in its multicategory: so for example and event that had mpp DY will end up in the DY multiclass and for that multiclassclass the DY output score will be used (which is already binned with correct mpp binning). It seems correct, but I might be missing something... :)
 
> Apart from this, I've provided you only the binnings for the 2016 data and I see you have applied them to all years. Do you want me to run it for the other years as well?

Yes please, if you could provide binnings for all years it would be great.

dzuolo commented:
Ciao @francescobrivio I had a look at the changes you made:
- sampleCfgs seems ok to me
- in the mainCfg_*_syst.cfg there should be a selection for each category_systUp/systDown (we already discussed this)
- in the mainCfg_*_limits.cfg there should be a variable for each DNN/Multiclass_systUp/systDown (see config/mainCfg_ETau_Legacy2017_limits.cfg)
- normal selectionCfgs, which are used with the mainCfg_*_limits.cfg, must contain the binning for all the DNN and Multiclass distributions. For 2017 the binning for DNN distributions is already defined and is 20,0,1. The one for the Multiclass must be added
- selection_*_syst.cfg files seems ok to me

francescobrivio commented:
@jaimeleonh please have a look at 0b1f902 to confirm all binnings are fine! thanks!

jaimeleonh commented:
> 
> 
> @jaimeleonh please have a look at [0b1f902](https://github.com/LLRCMS/KLUBAnalysis/commit/0b1f90207a615b2c54acd13fb11c840342f5681a) to confirm all binnings are fine! thanks!

Hi @francescobrivio, I think they are fine, yes!

francescobrivio commented:
>     * in the mainCfg_*_syst.cfg there should be a selection for each category_systUp/systDown (we already discussed this)

addressed in 52c3de2
>     * in the mainCfg_*_limits.cfg there should be a variable for each DNN/Multiclass_systUp/systDown (see config/mainCfg_ETau_Legacy2017_limits.cfg)

addressed in 1001c0c
>     * normal selectionCfgs, which are used with the mainCfg_*_limits.cfg, must contain the binning for all the DNN and Multiclass distributions. For 2017 the binning for DNN distributions is already defined and is 20,0,1. The one for the Multiclass must be added

addressed in 158d93e

francescobrivio commented:
I will merge this PR and the remaining points will be addressed in a future PR.

dzuolo commented:
@francescobrivio There is a mistake in the selectionCfgs: in some cases baselineVBFloose is defined starting from baseline, in some other cases starting from baseline55

francescobrivio commented:
> baselineVBFloose

Thanks for spotting that! addressed in #170
